# <font face = 'Palatino Linotype' color = '#6E5E5E'> 🥷 Master the Art of Linear Programming for Optimization Problem <font/>
## <font face = 'Palatino Linotype' color = '#7BA388'> Author and Trainer: Paolo G. Hilado MSc. (Data Science) <font/>

Step into the shadows of optimization and unleash the secret techniques of resource allocation!
In this notebook, we slice through constraints and stealthily maximize profits using Python’s PuLP library.
From crafting the perfect balance between time, materials, and output to solving real-world production puzzles, you’ll see how linear programming turns chaos into a razor-sharp strategy.

## <font face = 'Palatino Linotype' color = '#EB9830'> Scenario <font/>

SwiftFurniture Co. manufactures chairs and tables. Each product generates profit, but the factory has limited resources for wood and labor. To stay competitive, the company wants to maximize total profit while ensuring resources aren’t exceeded. This calls for linear programming for optimization.

## <font face = 'Palatino Linotype' color = '#EB9830'> Objective <font/>
Determine how many chairs and tables the factory should produce to maximize total profit, without exceeding the available wood and labor.

## <font face = 'Palatino Linotype' color = '#574407'> Business Understanding <font/>
### <font face = 'Palatino Linotype' color = '#EB9830'> Profit per unit: 20 USD for Chair & 30 USD for Table <font/>
### <font face = 'Palatino Linotype' color = '#574407'> Resource usage: <font/>

|    Resource   | Chair | Table | Total Available |
| ------------- | ----- | ----- | --------------- |
| Wood (units)  |   2   |   4   |   100           |
| Labor (hours) |   3   |   2   |   90            |

In [ ]:
# Install needed python library
!pip install pulp

In [1]:
# Import necessary python libraries
import pandas as pd
import numpy as np
from pulp import LpMaximize, LpProblem, LpVariable, value

In [2]:
# Establish the Linear Programming Problem
prob = LpProblem("Maximize_Profit", LpMaximize)

In [4]:
# Define decision variables: number of chairs and tables
chairs = LpVariable("Chairs", lowBound=0, cat='Continuous')
tables = LpVariable("Tables", lowBound=0, cat='Continuous')

In [5]:
# Indicate Objective function: Maximize profit
prob += 20 * chairs + 30 * tables, "Total_Profit"

In [7]:
# Indicate Constraints
prob += 2 * chairs + 4 * tables <= 100, "Wood_Constraint"
prob += 3 * chairs + 2 * tables <= 90, "Labor_Constraint"

In [8]:
# Solve the problem
prob.solve()
# Status 1 means solution is found

1

In [9]:
# Show results
print("Status:", prob.status)
print("Optimal number of chairs:", value(chairs))
print("Optimal number of tables:", value(tables))
print("Maximum Profit:", value(prob.objective))

Status: 1
Optimal number of chairs: 20.0
Optimal number of tables: 15.0
Maximum Profit: 850.0


## Given the Profit per unit and Resource Usage for making chairs and tables, linear programming informs us that we can maximize profit to 850 by creating 20 chairs and 15 tables. 